client/components/CheckoutTool/CheckoutTool.js

```diff
@@ -15,23 +15,42 @@ class CheckoutTool extends React.Component {
     };
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
im a little confused by this code/im unfamiliar with componentDidUpdate. i was thinking you do not need to store state anywhere other than App.jsx. you can pass event handlers to your child components that update state in App.jsx.

i think the way you have it setup makes it a little complex. for example, i assume you'll need checkInDate and checkOutDate for the DisplayCalendar component. how would this setup allow DisplayCalendar to have access to checkInDate or checkOutDate?

Reply:
I'm not entirely sure how else to do it because my child component takes in its values as props that are passed down.

I tried to implement it without storing its own local state, but I don't believe component will update will work // didn't investigate enough.

Essentially since I'm taking the data as props I only assign the props for my child component once when it renders but there's no way to tell the props to update once the state of the parent component changes. The componentWillUpdate method tests if the prev props differ from the current props and then I update the state which allows me to update the values in my child component.

Without it I think it would just be set as the original declaration of '' since we have no way of reassigning it

Thoughts? @maxmilpro

Reply:
i would take a look at the react docs about [lifting state up](https://reactjs.org/docs/lifting-state-up.html). their example is a little confusing. might be more straightforward examples out there.

happy to chat on zoom tomorrow too 

at a high level it would involve something like:

- move all state to parent component, App.jsx
- create event handler in parent component that updates state
- pass event handler to child component
- create onChange event handler in child component that uses event handler from parent component to update state